include/aws/s3/private/s3_auto_ranged_get.h

```diff
@@ -0,0 +1,38 @@
+#ifndef AWS_S3_AUTO_RANGED_GET_H
+#define AWS_S3_AUTO_RANGED_GET_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include "aws/s3/private/s3_meta_request_impl.h"
+
+enum aws_s3_auto_ranged_get_request_type {
+    AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_PART,
+};
+
+struct aws_s3_auto_ranged_get {
+    struct aws_s3_meta_request base;
+
+    struct {
```

Comment:
Where da mutex at?

Reply:
ok after reading further these seem to be locked higher up. A comment next to each one explaining that would do wonders.